### PR TITLE
[react] change sort icons for usermanagement

### PR DESCRIPTION
**Pull request by solnone**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

https://youtu.be/d8gXjk7k4xw

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 1]
```diff
@@ -44,6 +45,21 @@ export const UserManagement = () => {
...
+      login: faSort,
+      email: faSort,
+      langKey: faSort,
+      createdDate: faSort,
+      lastModifiedBy: faSort,
+      lastModifiedDate: faSort,
+    }
+    icons[sort] = sortUp ? faSortUp : faSortDown;
+    return icons;
+  };
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:60]:
```suggestion
  const createSortIcons = (sortUp: boolean, sort: string) => ({ [sort]: sortUp ? faSortUp : faSortDown });
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 2]
```diff
@@ -124,21 +143,21 @@ export const UserManagement = () => {
       <Table responsive striped>
         <thead>
           <tr>
-            <th className="hand" onClick={sort('id')}><Translate contentKey="global.field.id">ID</Translate><FontAwesomeIcon icon="sort" /></th>
-            <th className="hand" onClick={sort('login')}><Translate contentKey="userManagement.login">Login</Translate><FontAwesomeIcon icon="sort" /></th>
-            <th className="hand" onClick={sort('email')}><Translate contentKey="userManagement.email">Email</Translate><FontAwesomeIcon icon="sort" /></th>
+            <th className="hand" onClick={sort('id')}><Translate contentKey="global.field.id">ID</Translate><FontAwesomeIcon icon={sortIcons.id} /></th>
+            <th className="hand" onClick={sort('login')}><Translate contentKey="userManagement.login">Login</Translate><FontAwesomeIcon icon={sortIcons.login} /></th>
+            <th className="hand" onClick={sort('email')}><Translate contentKey="userManagement.email">Email</Translate><FontAwesomeIcon icon={sortIcons.email} /></th>
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:148]:
```suggestion
            <th className="hand" onClick={sort('id')}><Translate contentKey="global.field.id">ID</Translate><FontAwesomeIcon icon={sortIcons.id ?? faSort} /></th>
            <th className="hand" onClick={sort('login')}><Translate contentKey="userManagement.login">Login</Translate><FontAwesomeIcon icon={sortIcons.login ?? faSort} /></th>
            <th className="hand" onClick={sort('email')}><Translate contentKey="userManagement.email">Email</Translate><FontAwesomeIcon icon={sortIcons.email ?? faSort} /></th>
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 3]
```diff
@@ -124,21 +143,21 @@ export const UserManagement = () => {
...
             <th />
 <%_ if (enableTranslation) { _%>
-            <th className="hand" onClick={sort('langKey')}><Translate contentKey="userManagement.langKey">Lang Key</Translate><FontAwesomeIcon icon="sort" /></th>
+            <th className="hand" onClick={sort('langKey')}><Translate contentKey="userManagement.langKey">Lang Key</Translate><FontAwesomeIcon icon={sortIcons.langKey} /></th>
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:151]:
```suggestion
            <th className="hand" onClick={sort('langKey')}><Translate contentKey="userManagement.langKey">Lang Key</Translate><FontAwesomeIcon icon={sortIcons.langKey ?? faSort} /></th>
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 4]
```diff
@@ -124,21 +143,21 @@ export const UserManagement = () => {
...
 <%_ } _%>
             <th><Translate contentKey="userManagement.profiles">Profiles</Translate></th>
 <%_ if (!databaseTypeCassandra) { _%>
-            <th className="hand" onClick={sort('createdDate')}><Translate contentKey="userManagement.createdDate">Created Date</Translate><FontAwesomeIcon icon="sort" /></th>
+            <th className="hand" onClick={sort('createdDate')}><Translate contentKey="userManagement.createdDate">Created Date</Translate><FontAwesomeIcon icon={sortIcons.createdDate} /></th>
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:155]:
```suggestion
            <th className="hand" onClick={sort('createdDate')}><Translate contentKey="userManagement.createdDate">Created Date</Translate><FontAwesomeIcon icon={sortIcons.createdDate ?? faSort} /></th>
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 5]
```diff
@@ -124,21 +143,21 @@ export const UserManagement = () => {
...
             <th className="hand" onClick={sort('lastModifiedBy')}>
-              <Translate contentKey="userManagement.lastModifiedBy">Last Modified By</Translate><FontAwesomeIcon icon="sort" />
+              <Translate contentKey="userManagement.lastModifiedBy">Last Modified By</Translate><FontAwesomeIcon icon={sortIcons.lastModifiedBy} />
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:157]:
```suggestion
              <Translate contentKey="userManagement.lastModifiedBy">Last Modified By</Translate><FontAwesomeIcon icon={sortIcons.lastModifiedBy ?? faSort} />
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 6]
```diff
@@ -124,21 +143,21 @@ export const UserManagement = () => {
...
             </th>
             <th id="modified-date-sort" className="hand" onClick={sort('lastModifiedDate')}>
-              <Translate contentKey="userManagement.lastModifiedDate">Last Modified Date</Translate><FontAwesomeIcon icon="sort" />
+              <Translate contentKey="userManagement.lastModifiedDate">Last Modified Date</Translate><FontAwesomeIcon icon={sortIcons.lastModifiedDate} />
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:160]:
```suggestion
              <Translate contentKey="userManagement.lastModifiedDate">Last Modified Date</Translate><FontAwesomeIcon icon={sortIcons.lastModifiedDate ?? faSort} />
```

**Review by mshima — COMMENTED** [review]
What do you think about this?
It should make `createSortIcons` more generic.

**Review by mshima — DISMISSED** [review]

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs [hunk 7]
```diff
@@ -336,13 +343,13 @@ export const <%= entityReactName %> = () => {
               <thead>
                 <tr>
 <%_ for (field of fields) { _%>
-                  <th<% if (!paginationNo) { %> className="hand" onClick={sort('<%= field.fieldName %>')} <%_ } _%>><Translate contentKey="<%= `${i18nKeyPrefix}.${field.fieldName}` %>"><%= field.fieldNameHumanized %></Translate><% if (!paginationNo) { %> <FontAwesomeIcon icon="sort" /><% } %></th>
+                  <th<% if (!paginationNo) { %> className="hand" onClick={sort('<%= field.fieldName %>')} <%_ } _%>><Translate contentKey="<%= `${i18nKeyPrefix}.${field.fieldName}` %>"><%= field.fieldNameHumanized %></Translate><% if (!paginationNo) { %> <FontAwesomeIcon icon={sortIcons.<%= field.fieldName %>  ?? faSort} /><% } %></th>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs:346]:
```suggestion
                  <th<% if (!paginationNo) { %> className="hand" onClick={sort('<%= field.fieldName %>')} <%_ } _%>><Translate contentKey="<%= `${i18nKeyPrefix}.${field.fieldName}` %>"><%= field.fieldNameHumanized %></Translate><% if (!paginationNo) { %> <FontAwesomeIcon icon={sortIcons.<%= field.fieldName %> ?? faSort} /><% } %></th>
```

**Comment by solnone** [comment]
> I've done some design changes to angular sort icons recently, you may be interested #19371.

How about using text-nowrap
<th class="hand text-nowrap">

**Pull request closed without merging**